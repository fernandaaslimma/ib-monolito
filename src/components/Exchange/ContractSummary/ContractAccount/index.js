import ContractAccount from "./ContractAccount";

export default ContractAccount;
