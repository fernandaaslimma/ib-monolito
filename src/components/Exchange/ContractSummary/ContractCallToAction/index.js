import { connect } from "redux-zero/react";

import contractsActions from "../../../../actions/contracts";
import ContractCallToAction from "./ContractCallToAction";

export default connect(
  ({ signLoadingId }) => ({ signLoadingId }),
  contractsActions
)(ContractCallToAction);
