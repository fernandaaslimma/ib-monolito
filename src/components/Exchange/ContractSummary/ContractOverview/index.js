import ContractOverview from "./ContractOverview";

export default ContractOverview;
