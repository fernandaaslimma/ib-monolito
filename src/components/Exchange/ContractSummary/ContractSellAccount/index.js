import ContractSellAccount from "./ContractSellAccount";

export default ContractSellAccount;
