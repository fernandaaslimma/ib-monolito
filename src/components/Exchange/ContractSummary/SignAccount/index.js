import SignAccount from "./SignAccount";

export default SignAccount;
