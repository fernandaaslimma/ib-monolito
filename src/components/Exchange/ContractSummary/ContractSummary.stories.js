import React from "react";
import { storiesOf } from "@storybook/react"; // eslint-disable-line
import { withInfo } from "@storybook/addon-info"; // eslint-disable-line

import ContractSummary from "./ContractSummary";

const ContractSummaryProps = {
  contractId: "21332112XX",
  foreignAmount: 3212312,
  foreignCurrency: "USD",
  fxNature: "0921810923",
  localAmount: 32132232,
  localCurrency: "BRL",
  tradeDate: "20/12/2017",
  type: "buy",
  totalEffectiveRate: 2.333,
  destinationAccount: {
    account: "213223",
    bank: {
      aba: "2312",
      chips: "231",
      name: "BBM",
      swift: "333"
    },
    country: "China",
    name: "Bruce Lee"
  },
  intermediaryAccount: {
    account: "13231",
    bank: {
      aba: "567",
      chips: "657",
      name: "BBM",
      swift: "321"
    }
  },
  signUrl: "/",
  requestAjustmentUrl: "/"
};

storiesOf("Exchange/ContractSummary", module)
  .add(
    "empty",
    withInfo()(() => <ContractSummary requestAjustmentUrl="/" signUrl="/" />)
  )
  .add("buy", withInfo()(() => <ContractSummary {...ContractSummaryProps} />))
  .add(
    "sell",
    withInfo()(() => <ContractSummary {...ContractSummaryProps} type="sell" />)
  );
