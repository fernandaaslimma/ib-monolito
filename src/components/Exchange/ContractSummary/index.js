import { connect } from "redux-zero/react";

import ContractSummary from "./ContractSummary";
import contractsActions from "../../../actions/contracts";

export default connect(
  ({ groupsInProgress, userInfoMail }) => ({ groupsInProgress, userInfoMail }),
  contractsActions
)(ContractSummary);
