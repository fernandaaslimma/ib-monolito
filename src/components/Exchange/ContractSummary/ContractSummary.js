import React, { Component } from "react";
import { string, number, shape, bool, arrayOf, func } from "prop-types";

import { Title, SummaryWrapper, SummaryRow } from "./styles";
import Card from "../../common/Card";
import ContractOverview from "./ContractOverview";
import ContractCallToAction from "./ContractCallToAction";
import ContractAccount from "./ContractAccount";
import SignAccount from "./SignAccount";
import ContractSellAccount from "./ContractSellAccount";

import { translate } from "../../../utils/i18n";
import { BUY } from "../../../utils/constants";
import { orange10 } from "../../../styles/settings";
import { formatDateToLocale } from "../../../utils/formatDate";

class ContractSummary extends Component {
  render() {
    const {
      contractId,
      foreignAmount,
      foreignCurrency,
      fxNature,
      localAmount,
      localCurrency,
      tradeDate,
      type,
      rate,
      destinationAccount,
      intermediaryAccount,
      groups,
      loading,
      groupsInProgress,
      userInfoMail,
      getUserInfo
    } = this.props;
    getUserInfo && getUserInfo();
    if (loading) {
      return (
        <SummaryWrapper data-test={"ContractSummaryLoading"}>
          <Title>{translate("FX_CONTRACT")}</Title>
          <SummaryRow operation>
            <Card title={translate("OPERATION_SUMMARY")} icon="Cash">
              <ContractOverview loading={loading} />
            </Card>
            <Card title={translate("DESTINATION_ACCOUNT")} icon="CashTransfer">
              <ContractSellAccount loading={loading} />
            </Card>
          </SummaryRow>
        </SummaryWrapper>
      );
    }
    const parsedSettlementDate = formatDateToLocale(tradeDate);

    // TODO: Change validation after contract type is defined
    const isBuy = typeof type === "string" && type.toLowerCase().match(BUY);
    if (!type) {
      return null;
    }
    return (
      <SummaryWrapper data-test={`ContractSummary-${contractId}`}>
        <Title>
          {translate("FX_CONTRACT")}: {contractId}
        </Title>
        <SummaryRow operation>
          <Card title={translate("OPERATION_SUMMARY")} icon="Cash">
            <ContractOverview
              typeOfExchange={type}
              settlementDate={parsedSettlementDate}
              localCurrency={localCurrency}
              foreignCurrency={foreignCurrency}
              valueOfForeignCurrency={foreignAmount}
              valueOfNationalCurrency={localAmount}
              natureOfExchange={fxNature}
              exchangeRate={rate}
              data-test={`typeOfContract-${type}`}
            />
          </Card>
          {isBuy && destinationAccount && (
            <Card title={translate("DESTINATION_ACCOUNT")} icon="CashTransfer">
              <ContractSellAccount
                bank={destinationAccount.bank.name}
                owner={destinationAccount.name}
                bankId={destinationAccount.bank.id}
                agency={destinationAccount.branch}
                account={destinationAccount.account}
                document={destinationAccount.taxId}
              />
            </Card>
          )}
        </SummaryRow>

        {!isBuy && (
          <SummaryRow>
            {destinationAccount && (
              <Card
                title={translate("DESTINATION_ACCOUNT")}
                icon="CashTransfer"
              >
                <ContractAccount
                  bank={destinationAccount.bank.name}
                  account={destinationAccount.account}
                  country={destinationAccount.country}
                  owner={destinationAccount.name}
                  swift={destinationAccount.bank.swift}
                  chips={destinationAccount.bank.chips}
                  aba={destinationAccount.bank.aba}
                />
              </Card>
            )}
            {intermediaryAccount && (
              <Card title={translate("INTERMEDIARY_ACCOUNT")} icon="Bank">
                <ContractAccount
                  bank={intermediaryAccount.bank.name}
                  account={intermediaryAccount.account}
                  swift={intermediaryAccount.bank.swift}
                  chips={intermediaryAccount.bank.chips}
                  aba={intermediaryAccount.bank.aba}
                />
              </Card>
            )}
          </SummaryRow>
        )}
        {groups.length > 0 && (
          <SummaryRow>
            {destinationAccount && (
              <Card
                title={translate("SIGNATURES")}
                titleColor={orange10}
                iconContainerBg={"223, 149, 68"}
                icon="Signatures"
                iconColor={orange10}
              >
                <SignAccount
                  groups={groups}
                  contractId={contractId}
                  groupsInProgress={groupsInProgress}
                  email={userInfoMail}
                />
              </Card>
            )}
          </SummaryRow>
        )}
        {groups.length === 0 && (
          <Card>
            <ContractCallToAction contractId={contractId} />
          </Card>
        )}
      </SummaryWrapper>
    );
  }
}

ContractSummary.defaultProps = {
  groupsInProgress: [],
  loading: false,
  contractId: null,
  type: null,
  tradeDate: null,
  localCurrency: null,
  foreignCurrency: null,
  localAmount: null,
  foreignAmount: null,
  fxNature: null,
  totalEffectiveRate: null,
  destinationAccount: {
    bank: {}
  },
  intermediaryAccount: {
    bank: {}
  },
  rolePeople: null,
  groups: []
};

ContractSummary.propTypes = {
  groupsInProgress: arrayOf(string),
  loading: bool,
  contractId: string,
  foreignAmount: number,
  foreignCurrency: string,
  fxNature: string,
  localAmount: number,
  localCurrency: string,
  tradeDate: string,
  type: string,
  totalEffectiveRate: number,
  destinationAccount: shape({
    account: string,
    bank: shape({
      aba: string,
      chips: string,
      name: string,
      swift: string
    }),
    country: string,
    name: string
  }),
  rolePeople: string,
  intermediaryAccount: shape({
    account: string,
    bank: shape({
      aba: string,
      chips: string,
      name: string,
      swift: string
    })
  }),
  groups: arrayOf(
    shape({
      id: string,
      name: string,
      recipients: arrayOf(
        shape({
          email: string,
          embedded: bool,
          name: string,
          status: string,
          type: string
        })
      ),
      signOrder: number,
      status: string
    })
  ),
  resetSignLoading: func.isRequired
};

export default ContractSummary;
