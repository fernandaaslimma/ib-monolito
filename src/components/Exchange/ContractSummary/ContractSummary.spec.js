import React from "react";
import { shallow } from "enzyme";

import ContractSummary from "./ContractSummary";

const ContractSummaryProps = {
  contractId: "21332112XX",
  foreignAmount: 3212312,
  foreignCurrency: "USD",
  fxNature: "0921810923",
  localAmount: 32132232,
  localCurrency: "BRL",
  tradeDate: "20/12/2017",
  type: "buy",
  totalEffectiveRate: 2.333,
  destinationAccount: {
    account: "213223",
    bank: {
      aba: "2312",
      chips: "231",
      name: "BBM",
      swift: "333"
    },
    country: "China",
    name: "Bruce Lee"
  },
  intermediaryAccount: {
    account: "13231",
    bank: {
      aba: "567",
      chips: "657",
      name: "BBM",
      swift: "321"
    }
  },
  signUrl: "/",
  requestAjustmentUrl: "/"
};

jest.mock("../../../utils/i18n", () => ({
  translate: id => id,
  getLanguage: () => "en-US",
  isZhCN: () => false
}));

describe("ContractSummary component", () => {
  it("should match snapshot", () => {
    expect(
      shallow(
        <ContractSummary
          {...ContractSummaryProps}
          requestAjustmentUrl="/"
          signUrl="/"
        />
      )
    ).toMatchSnapshot();
  });

  it("should match snapshot as a buy summary", () => {
    expect(
      shallow(<ContractSummary {...ContractSummaryProps} />)
    ).toMatchSnapshot();
  });

  it("should match snapshot as a sell summary", () => {
    expect(
      shallow(<ContractSummary {...ContractSummaryProps} type="sell" />)
    ).toMatchSnapshot();
  });
});
