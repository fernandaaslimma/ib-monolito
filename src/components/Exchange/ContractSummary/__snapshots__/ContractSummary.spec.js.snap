// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContractSummary component should match snapshot 1`] = `
<styled.section
  data-test="ContractSummary-21332112XX"
>
  <styled.h1>
    FX_CONTRACT
    : 
    21332112XX
  </styled.h1>
  <styled.div
    operation={true}
  >
    <Card
      icon="Cash"
      iconColor={null}
      title="OPERATION_SUMMARY"
      titleColor={null}
    >
      <ContractOverview
        data-test="typeOfContract-buy"
        exchangeRate={null}
        foreignCurrency="USD"
        loading={false}
        localCurrency="BRL"
        natureOfExchange="0921810923"
        settlementDate="6/1/2010"
        typeOfExchange="buy"
        valueOfForeignCurrency={3212312}
        valueOfNationalCurrency={32132232}
      />
    </Card>
    <Card
      icon="CashTransfer"
      iconColor={null}
      title="DESTINATION_ACCOUNT"
      titleColor={null}
    >
      <ContractSellAccount
        account="213223"
        agency={null}
        bank="BBM"
        bankId={null}
        document={null}
        loading={null}
        owner="Bruce Lee"
      />
    </Card>
  </styled.div>
  <Card
    icon={null}
    iconColor={null}
    title={null}
    titleColor={null}
  >
    <ConnectWrapper
      contractId="21332112XX"
    />
  </Card>
</styled.section>
`;

exports[`ContractSummary component should match snapshot as a buy summary 1`] = `
<styled.section
  data-test="ContractSummary-21332112XX"
>
  <styled.h1>
    FX_CONTRACT
    : 
    21332112XX
  </styled.h1>
  <styled.div
    operation={true}
  >
    <Card
      icon="Cash"
      iconColor={null}
      title="OPERATION_SUMMARY"
      titleColor={null}
    >
      <ContractOverview
        data-test="typeOfContract-buy"
        exchangeRate={null}
        foreignCurrency="USD"
        loading={false}
        localCurrency="BRL"
        natureOfExchange="0921810923"
        settlementDate="6/1/2010"
        typeOfExchange="buy"
        valueOfForeignCurrency={3212312}
        valueOfNationalCurrency={32132232}
      />
    </Card>
    <Card
      icon="CashTransfer"
      iconColor={null}
      title="DESTINATION_ACCOUNT"
      titleColor={null}
    >
      <ContractSellAccount
        account="213223"
        agency={null}
        bank="BBM"
        bankId={null}
        document={null}
        loading={null}
        owner="Bruce Lee"
      />
    </Card>
  </styled.div>
  <Card
    icon={null}
    iconColor={null}
    title={null}
    titleColor={null}
  >
    <ConnectWrapper
      contractId="21332112XX"
    />
  </Card>
</styled.section>
`;

exports[`ContractSummary component should match snapshot as a sell summary 1`] = `
<styled.section
  data-test="ContractSummary-21332112XX"
>
  <styled.h1>
    FX_CONTRACT
    : 
    21332112XX
  </styled.h1>
  <styled.div
    operation={true}
  >
    <Card
      icon="Cash"
      iconColor={null}
      title="OPERATION_SUMMARY"
      titleColor={null}
    >
      <ContractOverview
        data-test="typeOfContract-sell"
        exchangeRate={null}
        foreignCurrency="USD"
        loading={false}
        localCurrency="BRL"
        natureOfExchange="0921810923"
        settlementDate="6/1/2010"
        typeOfExchange="sell"
        valueOfForeignCurrency={3212312}
        valueOfNationalCurrency={32132232}
      />
    </Card>
  </styled.div>
  <styled.div>
    <Card
      icon="CashTransfer"
      iconColor={null}
      title="DESTINATION_ACCOUNT"
      titleColor={null}
    >
      <ContractAccount
        aba="2312"
        account="213223"
        bank="BBM"
        chips="231"
        country="China"
        owner="Bruce Lee"
        swift="333"
      />
    </Card>
    <Card
      icon="Bank"
      iconColor={null}
      title="INTERMEDIARY_ACCOUNT"
      titleColor={null}
    >
      <ContractAccount
        aba="567"
        account="13231"
        bank="BBM"
        chips="657"
        country={null}
        owner={null}
        swift="321"
      />
    </Card>
  </styled.div>
  <Card
    icon={null}
    iconColor={null}
    title={null}
    titleColor={null}
  >
    <ConnectWrapper
      contractId="21332112XX"
    />
  </Card>
</styled.section>
`;
